dlp/inspect.js

```diff
@@ -192,56 +174,168 @@ function inspectGCSFile (authToken, bucketName, fileName, inspectConfig) {
+  dlp.createInspectOperation(request)
+    .then((createJobResponse) => {
+      const operation = createJobResponse[0];
+
+      // Start polling for job completion
+      return operation.promise();
+    })
+    .then((completeJobResponse) => {
+      // When job is complete, get its results
+      const jobName = completeJobResponse[0].name;
+      return dlp.listInspectFindings({
+        name: jobName
+      });
+    })
+    .then((results) => {
+      const findings = results[0].result.findings;
+      if (findings.length > 0) {
+        console.log(`Findings:`);
+        findings.forEach((finding) => {
+          console.log(`\tInfo type: ${finding.infoType.name}`);
+          console.log(`\tLikelihood: ${finding.likelihood}`);
+        });
+      } else {
+        console.log(`No findings.`);
+      }
+    })
+    .catch((err) => {
+      console.log('Error in promiseInspectGCSFile:', err.message || err);
+    });
+  // [END inspect_gcs_file_promise]
+}
+
+function eventInspectGCSFile (bucketName, fileName, minLikelihood, maxFindings, infoTypes) {
+  // [START inspect_gcs_file_event]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = DLP();
+
+  // The name of the bucket where the file resides.
+  // const bucketName = 'YOUR-BUCKET';
+
+  // The path to the file within the bucket to inspect.
+  // Can contain wildcards, e.g. "my-image.*"
+  // const fileName = 'my-image.png';
+
+  // The minimum likelihood required before returning a match
+  // const minLikelihood = LIKELIHOOD_UNSPECIFIED;
+
+  // The maximum number of findings to report (0 = server maximum)
+  // const maxFindings = 0;
+
+  // The infoTypes of information to match
+  // const infoTypes = ['US_MALE_NAME', 'US_FEMALE_NAME'];
+
+  // Get reference to the file to be inspected
+  const storageItems = {
+    cloudStorageOptions: {
+      fileSet: { url: `gs://${bucketName}/${fileName}` }
+    }
+  };
+
+  // Construct REST request body for creating an inspect job
+  const request = {
+    inspectConfig: {
+      infoTypes: infoTypes,
+      minLikelihood: minLikelihood,
+      maxFindings: maxFindings
     },
-    json: requestBody
+    storageConfig: storageItems
   };
 
-  // Run inspect-job creation REST request
-  requestPromise.post(options)
-    .then((createBody) => pollJob(createBody, inspectConfig.initialTimeout, inspectConfig.tries, authToken))
-    .then((jobName) => getJobResults(authToken, jobName))
-    .then((findingsBody) => {
-      const findings = findingsBody.result.findings;
-      console.log(JSON.stringify(findings, null, 2));
+  // Create a GCS File inspection job, and resolve it using event handlers
+  dlp.createInspectOperation(request)
+    .then((createJobResponse) => {
+      const operation = createJobResponse[0];
+      return new Promise((resolve, reject) => {
```

Comment:
This uses event handlers - and I wanted to avoid "callback hell".

(If you think the nested callback approach would be better idiomatically vs. mixing promises and event handlers, I'm happy to switch to it.)